compiler/rustc_mir_build/src/thir/pattern/const_to_pat.rs

```diff
@@ -191,8 +191,15 @@ impl<'tcx> ConstToPat<'tcx> {
                         }
                     } else {
                         let err = InvalidPattern { span: self.span, non_sm_ty };
-                        self.tcx().sess.emit_err(err);
-                        return Box::new(Pat { span: self.span, ty: cv.ty(), kind: PatKind::Wild });
+                        let e = self.tcx().sess.emit_err(err);
+                        let kind = PatKind::Constant {
+                            value: mir::ConstantKind::Ty(ty::Const::new_error(
```

Comment:
Oh I had missed the `new_error` here, that explains my confusion in the other thread.